### DISCUSSION TITLE
How to disable the dropdown of snippets after typing `s`?

### DISCUSSION THREAD

**Discussion by AlbertLei** [James-Yu/LaTeX-Workshop#3990] [Q&A]
**Problem Description.**
When I type `s` after a space, there will be a dropdown in the editor with alternatives "SCH", "SPA", .... 

<img width="613" alt="image" src="https://github.com/James-Yu/LaTeX-Workshop/assets/16741954/a7b42dec-3369-48d7-925f-56a05160a20a">

Similar dropdowns occur when I type `m`, `f`, `i`, ...

I find the dropdowns disturbing as they are not needed in more than 90 percent of my use cases.
To be clear, I am not asking how to disable code snippets (which I use a lot via tab completion), but how to disable the auto-intellisense (is that the right word?) of snippet.

Thanks in advance for any hint.

**Comment by jlelong** [comment]
The suggestions shown in the screenshot are coming from the snippets defined by the extension, see https://github.com/James-Yu/LaTeX-Workshop/wiki/Snippets

If you want to disable all snippets in latex documents and only keep intellisense, you can add the following code to your `settings.json` file (either User or workspace)

```
"latex": [
   { "editor.snippetSuggestions": "none" }
]
```

**Reply by AlbertLei** [reply]
It seems that `"editor.snippetSuggestions": "none"` disables snippets completely. I want to use snippets but just do not want to see the (too frequent) snippet Suggestions.

Is it possible to disable snippet **Suggestions** only? My real issue here is that the suggestion occurs every time I type a normal word beginning with letter s, i, b, m... 

Another possible workaround is to trigger snippet suggestions only when I type a Capital S, I, B, M,... That should reduce most unnecessary snippet suggestions. Is that possible?

thanks for your info.

**Reply by jlelong** [reply]
Unfortunately, LW cannot control how snippets are shown. VS Code snippets selection is performed case-insensitive, see https://github.com/microsoft/vscode/issues/139796

I am not sure to understand what you mean by 

> Is it possible to disable snippet Suggestions only?

Isn't it what setting `"editor.snippetSuggestions": "none"` does?

**Reply by AlbertLei** [reply]
> I am not sure to understand what you mean by "Is it possible to disable snippet Suggestions only?"

Setting `"editor.snippetSuggestions": "none"` disables all snippets provided by Latex-Workshop for me. Eg, when I type `...`+`TAB` or `BEQ`+`TAB`, it will not trigger `\dots` or the equation environment for me.

If possible, I wish to disable the snippet suggestions menu as in the screenshot and keep using snippets via tab completion.

**Reply by jlelong** [reply]
As far as I known it is impossible to disable the snippet popup menu and still use snippets via tab completion.
